internal/pkg/agent/application/paths/paths_windows.go

```diff
@@ -65,8 +65,13 @@ func initialControlSocketPath(topPath string) string {
 // RunningInstalled will always be false, even when it is an installed version. Once
 // that is fixed from the upgrade process the control socket path needs to be updated.
 func ResolveControlSocket() {
+	ResolveControlSocketWithInstalledState(RunningInstalled())
+}
+
+// ResolveControlSocketWithInstalledState updates the control socket path.
+func ResolveControlSocketWithInstalledState(runningInstalled bool) {
```

Comment:
Can we get some unit tests to maintain and document the behavior of `paths.ResolveControlSocketWithInstalledState(true)` ?  (we probably need to abstract the calls to `TopPath()` with a parameter and some overloading in order to use temp directories in test)